app/views/users/show.html.erb

```diff
@@ -4,7 +4,7 @@
   </div>
 </div>
 
-<% if current_user == @user || @user.private? || current_user.leaders.include?(@user) %>
+<% if policy(@user).show? %>
```

Comment:
Oh, I think I didn't call this policy here. I believe I did somewhere else, but maybe I should go back and adjust that quick!